docker/solr/entrypoint.sh

```diff
@@ -0,0 +1,213 @@
+# Since Solr 5.x
+SOLR_COLLECTION=${SOLR_COLLECTION:-gettingstarted}
+
+download() {
+    FILE="$2.tgz"
+    if [ -f $FILE ];
+    then
+       echo "File $FILE exists."
+       tar -zxf $FILE
+    else
+       echo "File $FILE does not exist. Downloading solr from $1..."
+       curl -O $1
+       tar -zxf $FILE
+    fi
+    echo "Downloaded!"
+}
+
+is_solr_up(){
+    echo "Checking if solr is up on http://localhost:$SOLR_PORT/solr/admin/cores"
+    http_code=`echo $(curl -s -o /dev/null -w "%{http_code}" "http://localhost:$SOLR_PORT/solr/admin/cores")`
+    return `test $http_code = "200"`
+}
+
+wait_for_solr(){
+    while ! is_solr_up; do
+        sleep 3
+    done
+}
+
+keep_solr_up(){
+    trap "exit" INT
+    while is_solr_up; do
+        echo "Still up!" 
+        sleep 2
+    done
+}
+
+run() {
+    dir_name=$1
+    solr_port=$2
+    solr_core=$3
+    # Run solr
+    echo "Running with folder $dir_name"
+    echo "Starting solr on port ${solr_port}..."
+
+    # go to the solr folder
+    cd $1/example
+
+    if [ "$DEBUG" = "true" ]
+    then
+        java -Djetty.port=$solr_port -Dsolr.solr.home=multicore -jar start.jar &
+    else
+        java -Djetty.port=$solr_port -Dsolr.solr.home=multicore -jar start.jar > /dev/null 2>&1 &
+    fi
+    wait_for_solr
+    cd ../../
+    echo "Started"
+}
+
+run_solr5_example() {
+    dir_name=$1
+    solr_port=$2
+    ./$dir_name/bin/solr -p $solr_port -c -e schemaless
+    echo "Started"
+}
+
+run_solr5() {
+    dir_name=$1
+    solr_port=$2
+    ./$dir_name/bin/solr -p $solr_port -c
+    echo "Started"
+}
+
+download_and_run() {
+	version=$1
+    case $1 in
+        3.*)
+            url="http://archive.apache.org/dist/lucene/solr/${version}/apache-solr-${version}.tgz"
+            dir_name="apache-solr-${version}"
+            dir_conf="conf/"
+            ;;
+        4.0.0)
+            url="http://archive.apache.org/dist/lucene/solr/4.0.0/apache-solr-4.0.0.tgz"
+            dir_name="apache-solr-4.0.0"
+            dir_conf="collection1/conf/"
+            ;;
+        4.*)
+            url="http://archive.apache.org/dist/lucene/solr/${version}/solr-${version}.tgz"
+            dir_name="solr-${version}"
+            dir_conf="collection1/conf/"
+            ;;
+        5.*|6.*)
+            url="http://archive.apache.org/dist/lucene/solr/${version}/solr-${version}.tgz"
+            dir_name="solr-${version}"
+            ;;
+        *)
+			echo "Sorry, $1 is not supported or not valid version."
+			exit 1
+    esac
+
+    download $url $dir_name
+    if [[ $1 == 5* || $1 == 6* ]]
+    then
+        if [ -z "${SOLR_COLLECTION_CONF}" ]
+        then
+            run_solr5_example $dir_name $SOLR_PORT
+        else
+            run_solr5 $dir_name $SOLR_PORT
+            create_collection $dir_name $SOLR_COLLECTION $SOLR_COLLECTION_CONF $SOLR_PORT
+        fi
+        if [ -z "${SOLR_DOCS}" ]
+        then
+            echo "SOLR_DOCS not defined, skipping initial indexing"
```

Comment:
IMO, you don't need this `SOLR_DOCS` stuff at all, as Packagist does not use it.